compiler/src/main/scala/org/scalajs/core/compiler/PreTyperComponent.scala

```diff
@@ -0,0 +1,121 @@
+/* Scala.js compiler
+ * Copyright 2013 LAMP/EPFL
+ * @author Nicolas Stucki
+ */
+
+package org.scalajs.core.compiler
+
+import scala.tools.nsc
+import nsc._
+
+/** This `jspretyper` phase prepares a fix for issue SI-9487 in the case of
+ *  anonymous classes that extend js.Any.
+ *
+ *  The issue SI-9487 transforms durring `typer` some public method definitions
```

Comment:
Also consider: During `typer`, Scalac transforms some [...] (SI-9487).
Or: Due to a bug (SI-9487), Scalac transfroms [...] during typer.